authority/provisioner/webhook_test.go

```diff
@@ -101,10 +101,10 @@ func TestWebhookController_isCertTypeOK(t *testing.T) {
 	}
 }
 
-// withRequestID is a helper that calls into [logging.WithRequestID] and returns
-// a new context with the requestID added to the provided context.
+// withRequestID is a helper that calls into [requestid.NewContext] and returns
+// a new context with the requestID added.
 func withRequestID(ctx context.Context, requestID string) context.Context {
```

Comment:
I think we should be injecting an ID in the context if none is present and, from what I understand, `withRequestID` is only called in tests atm.

The reason I'm mentioning this is that it'd be great if an outgoing call (to a webhook) contained a request ID in its logs, even though there's none attached to it from the caller.

Reply:
`withRequestID` is intended to be used in testing only. Have made that stronger by making it an actual helper. 

The request ID was already forwarded to all webhooks. It is guaranteed to exist and to be set, due to ordering of the CA middleware. I've added an additional integration test covering this case.